### PR TITLE
docs: update contribuing docs

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY

1. remove unused superset-ui section
2. fix dead link

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### TESTING INSTRUCTIONS

CI

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by villebro** [comment]
@riahk can you take a look at these changes? FYI @zhaoyongjie , Moriah is moving these sections into the docs section, so in the future the majority of this will be featured in the official documentation.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -728,7 +715,7 @@ def sqrt(x: Union[float, int]) -> Union[float, int]:
     return math.sqrt(x)
 ```
 
-### TypeScript
+### TypeScript Typing
```
Comment (riahk) [inline CONTRIBUTING.md:718]: This feels a bit redundant imo

**Review by riahk — APPROVED** [review]
Looks good! One slight nit. Thank you for this!

File: CONTRIBUTING.md (hunk 1 above)
Comment (zhaoyongjie) [reply CONTRIBUTING.md:718]: The `Typescript in Linting` and `Typescript in Typescript` are in conflict, so the Table of Contents isn't `typescript` in `Linting` section


image

**Pull request merged**